yaroslav.nazarenko/HW_6/parse.rb

```diff
@@ -0,0 +1,37 @@
+require 'nokogiri'
+
+# Do general things
+class Document
+  def initialize
+    @doc = Nokogiri::HTML(open('index.html'))
+  end
+
+  def parse(tag)
+    @result = @doc.css(tag)
+    puts "Here is all <#{tag}> elements from your file"
+  end
+end
+
+# Puts links
+class Link < Document
+  def parse(tag)
+    super
+    puts @result
+    puts '<-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=->'
+  end
+end
+
+# Puts headers
+class Header < Document
+  def parse(tag)
+    super
+    puts @result
+    puts '<-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=->'
+  end
+end
+
+link = Link.new
+header = Header.new
+
+link.parse('a')
```

Comment:
if you send here param - you lose the overall idea of Header and Link class. Please remove param passing from here and make sure `link.parse` method returns links only and `header.parse` returns h2's